### ISSUE TITLE
Flag to let random perspective be area preserving

### ISSUE THREAD

**Issue by Parskatt** [kornia/kornia#1591]
## 🚀 Feature
I suggest a flag for random perspective, e.g. "area_preserving=True" that samples homographies such that the transformed image is on average the same size as the original.

One could change the random generation so that it is at least approximately the same area after e.g. by letting the mapping perturbations go in random directions, or use some post-processing.

## Motivation
The current random implementation:
https://github.com/kornia/kornia/blob/master/kornia/augmentation/random_generator/_2d/perspective.py#L61-L75 

Basically guarantees that the transformed points are inside the image. This is good if one doesnt want to miss anything in the image, but it induces a scale bias such that the images basically always shrink. This is not great for augmentation.

## Pitch
For me, the class is almost unusable since the random generation is not what I'm looking for. Basically, I have to subclass and overwrite the methods. I would much prefer if it was standard in Kornia.

**Comment by shijianjian** [comment]
@Parskatt It is an interesting feature. Please file a PR!

**Comment by Parskatt** [comment]
@shijianjian Haven't contributed here before, so might have missed something. Anyway, I made a WIP pull request in #1592 .

**Comment by shijianjian** [comment]
Closed by #1592
